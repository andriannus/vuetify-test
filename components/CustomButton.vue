<template>
  <div>
    <v-btn color="success" @click="sayHello">
      {{ text }}
    </v-btn>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api'

export default defineComponent({
  name: 'CustomButton',

  props: {
    text: {
      type: String,
      default: '',
    },
  },

  setup() {
    function sayHello(): void {
      console.log('Hello')
    }

    return { sayHello }
  },
})
</script>
