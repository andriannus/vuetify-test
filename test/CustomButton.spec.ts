import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'

import CustomButton from '@/components/CustomButton.vue'

describe('CustomButton.vue', () => {
  let wrapper: Wrapper<Vue>
  let vuetify: Vuetify

  const localVue = createLocalVue()

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('should match snapshot', () => {
    wrapper = mount(CustomButton, {
      localVue,
      vuetify,
      propsData: {
        text: 'Hello',
      },
      stubs: ['v-btn']
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
