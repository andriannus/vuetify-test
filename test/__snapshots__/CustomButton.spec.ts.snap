// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton.vue should match snapshot 1`] = `
"<div>
  <v-btn-stub color=\\"success\\">
    Hello
  </v-btn-stub>
</div>"
`;
